Tests/Tests/ABTesting/ExPlatTests.swift

```diff
@@ -117,54 +108,119 @@ class ExPlatTests: XCTestCase {
         XCTAssertEqual(ExPlat.shared?.experimentNames, ["foo", "bar"])
     }
 
-    // Tests ExPlat anonymous configuration using TracksService.
-    //
-    func testTracksServiceAnonymousConfiguration() throws {
+#if os(iOS)
+    /// Tests the right assignments endpoint is called when ExPlat is configured through `TracksService`.
+    ///
+    func testAssignmentsEndpointWithAnonymousConfiguration() {
         // Given
-        let eventNamePrefix = "wooios"
-        let anonymousId = "123"
-        self.tracksService.platform = nil
-        self.tracksService.eventNamePrefix = eventNamePrefix
+        let expectation = XCTestExpectation(description: "The ExPlat assignments endpoint should have the right params")
+        let experiment = "test"
+        let anonId = "123"
+        let eventNamePrefix = "wooios_test"
+        let expectedEndpoint = Self.makeAssignmentsEndpoint(platform: eventNamePrefix, experiment: experiment, anonId: anonId)
+        stub(condition: isAbsoluteURLString(expectedEndpoint)) { request in
+            expectation.fulfill()
+            return .init(data: Data(), statusCode: 200, headers: nil)
+        }
 
         // When
-        self.tracksService.switchToAnonymousUser(withAnonymousID: anonymousId)
+        self.makeSharedExPlat(
+            platform: nil,
+            eventNamePrefix: eventNamePrefix,
+            experiment: experiment,
+            anonId: anonId
+        )
+        ExPlat.shared?.refresh()
 
         // Then
-#if os(iOS)
-        let exPlat = try XCTUnwrap(ExPlat.shared)
-        XCTAssertEqual(exPlat.platform, eventNamePrefix)
-        XCTAssertEqual(exPlat.oAuthToken, nil)
-        XCTAssertEqual(exPlat.anonId, anonymousId)
-#else
-        XCTAssertNil(ExPlat.shared)
-#endif
+        wait(for: [expectation], timeout: 1.0)
     }
 
-    // Tests ExPlat user authenticated configuration using TracksService.
-    //
-    func testTracksServiceUserAuthConfiguration() throws {
+    /// Tests the right assignments endpoint is called when ExPlat is configured through `TracksService`.
+    ///
+    func testAssignmentsEndpointWithUserAuthConfiguration() {
         // Given
-        let username = "foobar"
-        let userID = "123"
-        let wpComToken = "abc"
-        let platform = "wpios"
-        let eventNamePrefix = "jpios"
-        let skipAliasEventCreation = true
-        self.tracksService.platform = platform
-        self.tracksService.eventNamePrefix = eventNamePrefix
+        let expectation = XCTestExpectation(description: "The ExPlat assignments endpoint should have the right params")
+        let experiment = "test"
+        let platform = "wooios_test"
+        let token = "abc"
+        let expectedEndpoint = Self.makeAssignmentsEndpoint(platform: platform, experiment: experiment)
+        stub(condition: isAbsoluteURLString(expectedEndpoint)) { request in
```

Comment:
This condition is probably too strict. The query part in the `expectedEndpoint` probably won't exactly match the requested URL string. I think it's best to compare the path part in the condition, and assert the `request.url` contains certain query?

Reply:
See also [the method's implementation](https://github.com/AliSoftware/OHHTTPStubs/blob/12f19662426d0434d6c330c6974d53e2eb10ecd9/Sources/OHHTTPStubsSwift/OHHTTPStubsSwift.swift#L191).